### ISSUE TITLE
Problem backuping - Fatal: unable to save snapshot: Lstat: CreateFile

### ISSUE THREAD

**Issue by geralt18** [restic/restic#3339]
Output of `restic version`
--------------------------
restic 0.12.0 compiled with go1.15.8 on windows/amd64

How did you run restic exactly?
-------------------------------
```
restic -r \\192.168.0.250\backupDell\Serwis backup \\192.168.0.250\Serwis --exclude @Recently-Snapshot --exclude @Recycle

repository 4181a000 opened successfully, password is correct
no parent snapshot found, will read all files
Fatal: unable to save snapshot: Lstat: CreateFile \\?\c:\sql.backup\auto\192.168.0.250: Nie można odnaleźć określonego pliku.
/\\192.168.0.250\Serwis/192.168.0.250/Serwis/Zgk_Żagań/2021-03-15 WG Analizy i wydruki/v2.pdf
c:\sql.backup\auto>rwis/192.168.0.250/Serwis/Zgk_Żagań/2021-03-15 WG Analizy i wydruki/Analiza żagań 2021-03-16.xlsx

c:\sql.backup\auto>restic -r \\192.168.0.250\backupDell\Gw-Max backup \\192.168.0.250\gw-max --exclude @Recently-Snapshot --exclude @Recycle

repository 371ca448 opened successfully, password is correct
no parent snapshot found, will read all files
Fatal: unable to save snapshot: Lstat: CreateFile \\?\c:\sql.backup\auto\192.168.0.250: Nie można odnaleźć określonego pliku.
/\\192.168.0.250\gw-max/192.168.0.250/gw-max/tmp/tła_z_logiem/9.png
c:\sql.backup\auto>-max/192.168.0.250/gw-max/tmp/tła_z_logiem/Thumbs.db
```

What backend/server/service did you use to store the repository?
----------------------------------------------------------------
I use Windows Server 2019 to run restic.
Repository is on QNAP NAS. 

Expected behavior
-----------------
Add file to backup and create snapshot

Actual behavior
---------------
Error. Unable to create snapshot

Steps to reproduce the behavior
-------------------------------
No idea.

Do you have any idea what may have caused this?
-----------------------------------------------
Maybe file name with Polish letters?
But when I create files and dirs with Polish letters everything works fine.

Do you have an idea how to solve the issue?
-------------------------------------------
No

**Comment by MichaelEischer** [comment]
The error looks like something went wrong with mangling the repository path: ` CreateFile \\?\c:\sql.backup\auto\192.168.0.250` shouldn't be accessed at all. Do other command work for you? Does backing up a local folder work?

**Comment by geralt18** [comment]
Backing up local folder works fine. 
Backing up remote share (from my other NAS also works fine).

I can do check, prune, forget - I think they work properly.

Backing up path \\192.168.0.250\Serwis to local repository doesn't work - but error is on different file: 

```
restic -r repo backup \\192.168.0.250\Serwis --exclude @Recently-Snapshot --exclude @Recycle
enter password for repository:
repository 04fbf44b opened successfully, password is correct
created new cache in C:\Users\backup\AppData\Local\restic
no parent snapshot found, will read all files
Fatal: unable to save snapshot: Lstat: CreateFile \\?\c:\sql.backup\auto\192.168.0.250: Nie można odnaleźć określonego pliku.2.168.0.250\Serwis/192.168.0.250/Serwis/Świerklaniec/Polecenie wymiany wodomierza/cr-wod-wym.rpt
/\\192.168.0.250\Serwis/192.168.0.250/Serwis/!Tematy/DrukarkiMobilne/Zebra ZQ520/zsu-1191269.exe
```

**Comment by MichaelEischer** [comment]
I did a bit of testing and for me reading from and backing up to UNC paths works. So the next step would be to create a debug log to hopefully understand where that error originates from. Please see https://github.com/restic/restic/blob/master/CONTRIBUTING.md#reporting-bugs for a short explanation on creating a debug build. Feel free to ask if you need more detailed steps.

**Comment by geralt18** [comment]
Hi.
I mange to build restic - it was quite easy.
I attach log.
[restic-debug.log](https://github.com/restic/restic/files/6219723/restic-debug.log)

**Comment by MichaelEischer** [comment]
Thanks for the debug log and sorry for the long delay. The relevant log lines are the following ones:
```
2021/03/28 10:12:57 archiver/tree.go:292	archiver.NewTree	1	before unroll:
/\\192.168.0.250\Serwis, root ".", path "", meta "."
    /192.168.0.250, root "", path "", meta "192.168.0.250"
        /Serwis, root "", path "\\\\192.168.0.250\\Serwis", meta ""
```

That tree seems to be rather broken. Apparently the go standard library considers the path `\\192.168.0.250\Serwis` to be a relative path and not an absolute one. I'm not sure whether that is correct or not. According to the documentation at https://docs.microsoft.com/en-us/dotnet/standard/io/file-path-formats#unc-paths there are no relative UNC paths, which sounds like `\\192.168.0.250\Serwis` should be considered an absolute path. Is anyone more familiar with paths on windows and can help here?

Long story short, appending a single slash `\` should fix that. So `restic backup \\192.168.0.250\Serwis\` should work.  @geralt18 could you test whether that works? Please also test whether restoring is possible, I vaguely remember that there have been some issues related to UNC paths.

**Comment by geralt18** [comment]
Hi.
Appending a single slash `\` allowed me to create backup but restore is not working. 
I get the following error:
`
c:\restic>restic -r repo restore latest --target xxx
enter password for repository:
repository 82713f76 opened successfully, password is correct
restoring <Snapshot 986ecdc9 of [\\192.168.0.250\Serwis\] at 2021-05-18 09:18:56.887176 +0200 CEST by GIGA\backup@DELL> to xxx
ignoring error for \: invalid child node name \\192.168.0.250\Serwis
ignoring error for \: invalid child node name \\192.168.0.250\Serwis
Fatal: There were 2 errors
`

**Comment by ProactiveServices** [comment]
> According to the documentation at https://docs.microsoft.com/en-us/dotnet/standard/io/file-path-formats#unc-paths there are no relative UNC paths, which sounds like `\\192.168.0.250\Serwis` should be considered an absolute path. Is anyone more familiar with paths on windows and can help here?

The Microsoft documentation you linked matches my experience using UNC in the wild.

> Long story short, appending a single slash `\` should fix that. So `restic backup \\192.168.0.250\Serwis\` should work. 

Agreed, and suffixing a Windows path with a `\` will not affect its validity as far as Windows is concerned. Having tested, restic drops the trailing `\`.

[Restore of UNC Paths within windows failing](https://github.com/restic/restic/issues/2807) is related and still an issue - so you'll have a backup that cannot currently be restored.

**Comment by MichaelEischer** [comment]
I've opened issue https://github.com/golang/go/issues/47123 . Let's see what the Go devs think about it.

**Comment by underdpt** [comment]
That issue is closed, referencing https://go-review.googlesource.com/c/go/+/253197/

That's from March 2021, it seems the patch was merged (https://github.com/golang/go/commit/402d784b8fc0a7914955eb168ea288338e2025ff) but I cannot find it on the changelog (but it's merged on the go1.17 branch).

I did compile restic with latest go1.17.6 and the issue it's still there, so maybe there's still something to investigate. I'm willing to help (but I have no go knowledge, sorry).

**Comment by MichaelEischer** [comment]
I just gave this a try. This seems to no longer be a problem with recent restic versions, see also https://github.com/restic/restic/issues/2076 .

The restore issues for backups from UNC paths are tracked in https://github.com/restic/restic/issues/2807 .
